indico/core/notifications.py

```diff
@@ -167,11 +169,14 @@ def make_email(to_list=None, cc_list=None, bcc_list=None, from_address=None, rep
     cc_list = {cc_list} if isinstance(cc_list, str) else cc_list
     bcc_list = {bcc_list} if isinstance(bcc_list, str) else bcc_list
     reply_address = {reply_address} if (isinstance(reply_address, str) and reply_address) else (reply_address or set())
+    from_address = _get_actual_sender_address(sender_address)
+    if sender_address and from_address != sender_address:
+        reply_address |= {sender_address}
```

Comment:
I'm not sure if (even though the RFC allows it) using multiple reply-to addresses is a good idea.

I wonder whether we shouldn't clean up the API of `make_email` to be less ambiguous. One possibility would be to have only the `sender_address` arg, and completely get rid of the `reply_address` options - and then rely on the existing logic to determine whether the address goes in From or Reply-to.

Or alternatively, to make the PR less invasive, keep the API mostly as it is, but simply make an explicitly specified `reply_address` override the one that may otherwise be taken from the `from_address`.

I'll think about it and try around a bit on my dev instance...

Reply:
I have no strong opinion because in our case we use the reply_address option in make_email since we were converting all from_address to reply_address. But there are some notification emails functions which are explicitly passing the reply_address address to the make_email function. So maybe your second option would be better

Reply:
Hi @ThiefMaster, I just rebased this pr. Did you decide yet if I should revert the naming of `sender_address` back to `from_address`?